File: app/src/main/java/capstone/project/easycook/di/AppInjector.java
// Skeleton Injector that we define but Dagger will fill in for us
package capstone.project.easycook.di;

import javax.inject.Singleton;

import capstone.project.easycook.di.viewmodel.ViewModelModule;
import capstone.project.easycook.view.CookRecipeActivity;
import capstone.project.easycook.view.CreateRecipeActivity;
import capstone.project.easycook.view.RecipeActivity;
import capstone.project.easycook.view.RecipeListActivity;
import capstone.project.easycook.view.SelectCategoryActivity;
import dagger.Component;

// This will be able to inject the string module defined in this directory
@Component(modules = {AppModule.class, ViewModelModule.class})
@Singleton
public interface AppInjector
{
    void inject(SelectCategoryActivity activity);
    void inject(RecipeListActivity activity);
    void inject(CookRecipeActivity activity);
    void inject(CreateRecipeActivity activity);
    void inject(RecipeActivity activity);
}

File: app/src/main/java/capstone/project/easycook/di/AppInjector.kt
// Skeleton Injector that we define but Dagger will fill in for us
package capstone.project.easycook.di

import capstone.project.easycook.di.viewmodel.ViewModelModule
import capstone.project.easycook.view.*
import dagger.Component
import javax.inject.Singleton

// This will be able to inject the string module defined in this directory
@Singleton
@Component(modules = [AppModule::class, DatabaseModule::class, ViewModelModule::class])
interface AppInjector
{
    fun inject(activity: SelectCategoryActivity)
    fun inject(activity: RecipeListActivity)
    fun inject(activity: CookRecipeActivity)
    fun inject(activity: CreateRecipeActivity)
    fun inject(activity: RecipeActivity)

    fun inject(activity: DebugActivity)
}